runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5845,9 +5846,16 @@ class INTERPRETER_CLASS
 			break;
 		}
 		case J9VM_CONTINUATION_RETURN_FROM_SYNC_METHOD: {
-			UDATA *bp = ((UDATA *)(((J9SFMethodFrame *)_sp) + 1)) - 1;
-			restoreSpecialStackFrameLeavingArgs(REGISTER_ARGS, bp);
-			rc = inlineSendTarget(REGISTER_ARGS, VM_MAYBE, VM_MAYBE, VM_MAYBE, VM_MAYBE);
+			/* Reset interpreter state to what it would have been upon entry to inline send target */
+			J9SFStackFrame *bytecodeFrame = (J9SFStackFrame *)_sp;
+			J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(_literals);
+			_sp = (UDATA *)((J9SFStackFrame *)_sp + 1) + (romMethod->tempCount + 1);
+			_sendMethod = _literals;
+			_literals = bytecodeFrame->savedCP;
+			_pc = bytecodeFrame->savedPC;
+			bytecodeFrame->savedA0 = (UDATA*)((UDATA)bytecodeFrame->savedA0 & ~((UDATA)J9SF_A0_INVISIBLE_TAG));
```

Comment:
```suggestion
			/* Reset interpreter state to what it would have been upon entry to inline send target. */
			J9SFStackFrame *bytecodeFrame = (J9SFStackFrame *)_sp;
			J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(_literals);
			_sp = (UDATA *)((J9SFStackFrame *)_sp + 1) + (romMethod->tempCount + 1);
			_sendMethod = _literals;
			_literals = bytecodeFrame->savedCP;
			_pc = bytecodeFrame->savedPC;
			bytecodeFrame->savedA0 = (UDATA *)((UDATA)bytecodeFrame->savedA0 & ~((UDATA)J9SF_A0_INVISIBLE_TAG));
```